espresso/core/java/androidx/test/espresso/util/SpecialCharacter.java

```diff
@@ -0,0 +1,47 @@
+package androidx.test.espresso.util;
+
+import android.util.Pair;
+
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+
+import org.hamcrest.Description;
+
+import java.util.LinkedList;
+import java.util.List;
+
+public class SpecialCharacter {
+
+  private static final List<Pair<String, String>> SPECIAL_CHAR_CODES = new LinkedList<Pair<String, String>>() {{
+    add(new Pair<>("\u00A0", "[NBSP]"));
+  }};
+
+  @NonNull
+  public static String highlight(@NonNull CharSequence text) {
+    String result = text.toString();
+    for (Pair<String, String> entry : SPECIAL_CHAR_CODES) {
+      result = result.replace(entry.first, entry.second);
+    }
+    return result;
+  }
+
+  public static Description appendHighlightedValue(
```

Comment:
It does visualization. Besides, it's a nice helper method for other classes, who would like to use it in other matchers.
Since It also has a description referencing this class, I'm not sure we should move it to some other place.